docs/pages/router/reference/api-routes.mdx

```diff
@@ -196,6 +198,163 @@ export async function GET(request: Request, { post }: Record<string, string>) {
 
+## Runtime API
+
+> **important** The server runtime API and `expo-server` are available in SDK 54 and later, and require a deployed server for production use.
+
+You can use the `expo-server` package to use several utilities and code patterns that work in any server-side Expo code. This includes utilities to get request metadata, for scheduling tasks, and for error handling.
+
+<Terminal cmd={['$ npx expo install expo-server']} />
+
+Using `expo-server` is not limited to API routes and it can be used in any other server code as well, for example in [server middleware](/router/reference/middleware/).
+
+### Error handling
+
+You can abort a request and instead return an error `Response` by throwing a [`StatusError`](/versions/latest/sdk/server/#statuserror). This is a special `Error` instance that will be replaced with an HTTP response replacing the error itself.
+
+```ts app/blog/[post]+api.ts
+import { StatusError } from 'expo-server';
+
+export async function GET(request: Request, { post }: Record<string, string>) {
+  if (!post) {
+    throw new StatusError(404, 'No post found');
+  }
+  // ...
+}
+```
+
+When composing your own server utilities and helpers, the `StatusError` is a more convenient way to handle exceptions, since throwing them interrupts any API functions and instead returns an error early.
+
+`StatusError`s accept a status code and an error message, which can also optionally be passed as a JSON, or `Error` object, and will always return a `Response` with a JSON body with an `error` key set to their error message.
+
+This can be restrictive, and isn't suitable for all cases. Sometimes it might be beneficial to instead `throw` a `Response` object, which interrupts your logic as well, but replaces the resolved `Response` from your API route directly, without a `StatusError` wrapper.
+For example, this can be used to create redirect responses.
+
+```ts app/blog/[post]+api.ts
+import { StatusError } from 'expo-server';
+
+export async function GET(request: Request, { post }: Record<string, string>) {
+  if (!post) {
+    throw Response.redirect('https://expo.dev', 302);
+  }
+  // ...
+}
+```
+
+### Request metadata
+
+Requests typically carry most metadata you'll need in their headers. However, `expo-server` provides some helper functions to retrieve common values more easily.
+
+Helper functions from `expo-server` return values that are scoped to the current `Request`. You can only call these functions in server-side code and only during ongoing requests.
+
+A common value that you may need to access is the request's origin URL. The origin URL - typically transmitted on a request's `Origin` header, represents the URL that a user used to access your API route. This may differ from any internal deployment URL that your server sees when the request is being proxied. You can use `expo-server`'s [`origin()`](/versions/latest/sdk/server/#origin) helper to access this value.
```

Comment:
```suggestion
A common value that you may need to access is the request's origin URL. The origin URL, typically transmitted on a request's `Origin` header, represents the URL that a user used to access your API route. This may differ from any internal deployment URL that your server sees when the request is being proxied. You can use `expo-server`'s [`origin()`](/versions/latest/sdk/server/#origin) helper method to access this value.
```